engines/glk/advsys/game.cpp

```diff
@@ -176,7 +176,7 @@ void Game::loadGameData(Common::ReadStream &rs) {
 
 int Game::findWord(const Common::String &word) const {
 	// Limit the word to the maximum allowable size
-	Common::String w(word.c_str(), word.c_str() + WORD_SIZE);
+	Common::String w(word.c_str(), MIN(word.size(), (uint)WORD_SIZE));
```

Comment:
Shouldn't that be MAX? Since WORD_SIZE is 6, it'll always get picked, even for bigger words

```suggestion
	Common::String w(word.c_str(), MAX(word.size(), (uint)WORD_SIZE));
```